src/lune/builtins/datetime/mod.rs

```diff
@@ -0,0 +1,222 @@
+            LuaValue::Table(t) => Ok(DateTime::from_unix_timestamp(
+                TimestampType::Seconds,
+                t.get("unixTimestamp")?,
+            )),
+            _ => panic!("invalid type"),
+        }
+    }
+}
+
+impl LuaUserData for DateTimeBuilder {
+    fn add_fields<'lua, F: LuaUserDataFields<'lua, Self>>(fields: &mut F) {
+        fields.add_field_method_get("year", |_, this| Ok(this.year));
+        fields.add_field_method_get("month", |_, this| Ok(this.month));
+        fields.add_field_method_get("day", |_, this| Ok(this.day));
+        fields.add_field_method_get("hour", |_, this| Ok(this.hour));
+        fields.add_field_method_get("minute", |_, this| Ok(this.minute));
+        fields.add_field_method_get("second", |_, this| Ok(this.second));
+        fields.add_field_method_get("millisecond", |_, this| Ok(this.millisecond));
+    }
+}
+
+impl<'lua> FromLua<'lua> for DateTimeBuilder {
+    fn from_lua(value: LuaValue<'lua>, _: &'lua Lua) -> LuaResult<Self> {
+        match value {
+            LuaValue::Table(t) => Ok(Self::default()
+                .with_year(t.get("year")?)
+                .with_month(
+                    (match t.get("month")? {
+                        LuaValue::String(str) => Ok(str.to_str()?.parse::<Month>().or(Err(
+                            LuaError::external("could not cast month string to Month"),
+                        ))?),
+                        LuaValue::Nil => {
+                            Err(LuaError::external("cannot find mandatory month argument"))
+                        }
+                        LuaValue::Number(num) => Ok(Month::try_from(num as u8).or(Err(
+                            LuaError::external("could not cast month number to Month"),
+                        ))?),
+                        LuaValue::Integer(int) => Ok(Month::try_from(int as u8).or(Err(
+                            LuaError::external("could not cast month integer to Month"),
+                        ))?),
+                        _ => Err(LuaError::external("unexpected month field type")),
+                    })?,
+                )
+                .with_day(t.get("day")?)
+                .with_hour(t.get("hour")?)
+                .with_minute(t.get("minute")?)
+                .with_second(t.get("second")?)
+                .with_millisecond(t.get("millisecond")?)
+                // TODO: millisecond support
+                .build()),
+            _ => Err(LuaError::external(
+                "expected type table for DateTimeBuilder",
+            )),
+        }
+    }
+}
+
+impl<'lua> FromLua<'lua> for Timezone {
+    fn from_lua(value: LuaValue<'lua>, _: &'lua Lua) -> LuaResult<Self> {
+        fn num_to_enum(num: i32) -> LuaResult<Timezone> {
+            match num {
+                1 => Ok(Timezone::Utc),
+                2 => Ok(Timezone::Local),
+                _ => Err(LuaError::external("Invalid enum member!")),
+            }
+        }
```

Comment:
I dont think users being able to pass enums as plain numbers is a good idea. We should stick to string literals - they are much better for type checking / autocomplete, and it will be more obvious to readers what's going on.